protocol/app/ante/sigverify.go

```diff
@@ -89,23 +93,58 @@ func (svd SigVerificationDecorator) AnteHandle(
 		// Check account sequence number.
 		// Skip individual sequence number validation since this transaction use
 		// `GoodTilBlock` for replay protection.
-		if !skipSequenceValidation && sig.Sequence != acc.GetSequence() {
-			labels := make([]gometrics.Label, 0)
-			if len(tx.GetMsgs()) > 0 {
-				labels = append(
-					labels,
-					metrics.GetLabelForStringValue(metrics.MessageType, fmt.Sprintf("%T", tx.GetMsgs()[0])),
-				)
+		if !skipSequenceValidation {
+			if accountpluskeeper.IsTimestampNonce(sig.Sequence) {
+				tsNonce := sig.Sequence
+				blockTs := uint64(ctx.BlockTime().UnixMilli())
+				address := acc.GetAddress()
```

Comment:
Note that the `acc` and `sig` are not necessarily the same account (see [thread](https://dydx-team.slack.com/archives/C03FP255RRU/p1721848422208089)). May be helpful for readability to briefly explain why this is working as intended.